Objects/codeobject.c

```diff
@@ -1797,28 +1797,26 @@ code_richcompare(PyObject *self, PyObject *other, int op)
     for (int i = 0; i < Py_SIZE(co); i++) {
         _Py_CODEUNIT co_instr = _PyCode_CODE(co)[i];
         _Py_CODEUNIT cp_instr = _PyCode_CODE(cp)[i];
-        uint8_t co_code = co_instr.op.code;
+        uint8_t co_code = _Py_GetBaseOpcode(co, i);
         uint8_t co_arg = co_instr.op.arg;
-        uint8_t cp_code = cp_instr.op.code;
+        uint8_t cp_code = _Py_GetBaseOpcode(cp, i);
         uint8_t cp_arg = cp_instr.op.arg;
 
         if (co_code == ENTER_EXECUTOR) {
             const int exec_index = co_arg;
             _PyExecutorObject *exec = co->co_executors->executors[exec_index];
-            co_code = exec->vm_data.opcode;
+            co_code = _PyOpcode_Deopt[exec->vm_data.opcode];
```

Comment:
Nice catch. I think `deopt_code` should be doing this as well (it can't happen yet because these are all `JUMP_BACKWARD`s, which don't have specializations, but it could later when we start sticking executors in arbitrary locations).

I agree with @gvanrossum though that `_Py_GetBaseOpcode` should just be the one place where all of this logic is kept. @gaogaotiantian, would you be interested in teaching `_Py_GetBaseOpcode` about `ENTER_EXECUTOR` and cleaning up all of the places where we're using that and `_PyOpcode_Deopt` to do the right thing? I can do it if not.

Reply:
Yes, I can work on that.

Reply:
See the list of bullets in gh-107265.